base/float16.jl

```diff
@@ -1,11 +1,11 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 function convert(::Type{Float32}, val::Float16)
-    ival::UInt32 = reinterpret(UInt16, val)
```

Comment:
the code just looked silly with only local on the `ret` declation. I didn't deprecate this form, since it's not clear to me that is decided.